lib/genome/online_updaters/docm/updater.rb

```diff
@@ -0,0 +1,132 @@
+require 'net/http'
+module Genome; module OnlineUpdaters; module Docm
+  class Updater
+    attr_reader :new_version
+    def initialize(source_db_version = Date.today.strftime("%d-%B-%Y"))
+      @new_version = source_db_version
+    end
+
+    def update
+      remove_existing_source
+      source = create_new_source
+      create_interaction_claims(source)
+    end
+
+    private
+    def create_interaction_claims(source)
+      api_client = ApiClient.new
+      api_client.variants.each do |variant|
+        interaction_information = parse_interaction_information(variant)
+        interaction_information.each do |interaction_info|
+          gc = get_or_create_gene_claim(variant, source)
+          dc = get_or_create_drug_claim(interaction_info, source)
+          get_or_create_interaction_claim(gc, dc, interaction_info, variant['diseases'], source)
+        end
+      end
+    end
+
+    def get_or_create_gene_claim(variant, source)
+      DataModel::GeneClaim.where(
+        source_id: source.id,
+        nomenclature: 'DoCM Entrez Gene Symbol',
+        name: variant['gene']
+      ).first_or_create
+      #Entrez Gene Id
+    end
+
+    def parse_interaction_information(variant)
+      if !variant.key?('meta')
+        return []
+      end
+      fields = variant['meta'].first['Drug Interaction Data']['fields']
+      rows = variant['meta'].first['Drug Interaction Data']['rows']
+      interaction_information = Set.new()
+      rows.each do |row|
+        row_hash = {}
+        fields.zip(row).each do |name, value|
+          row_hash[name] = value
+        end
+        row_hash['Therapeutic Context'].split(/,|\+|plus/).each do |drug|
+          info = row_hash
+          info['Therapeutic Context'] = drug
+          if valid_drug?(drug)
+            interaction_information.add(info)
+          end
+        end
+      end
+      return interaction_information
+    end
+
+    def valid_drug?(drug_name)
+      [
+        !drug_name.include?('inhib'),
+        !drug_name.include?('HER3'),
+        !drug_name.include?('TKI'),
+        !drug_name.include?('anti'),
+        !drug_name.include?('BRAF'),
+        !drug_name.include?('radio'),
+        !drug_name.include?('BH3'),
+      ].all?
+    end
+
+    def get_or_create_drug_claim(interaction_info, source)
+      DataModel::DrugClaim.where(
+        source_id: source.id,
+        primary_name: interaction_info['Therapeutic Context'].upcase,
+        name: interaction_info['Therapeutic Context'].upcase,
+        nomenclature: 'DoCM Drug Name'
+      ).first_or_create
+    end
+
+    def get_or_create_interaction_claim(gene_claim, drug_claim, interaction_info, diseases, source)
+      DataModel::InteractionClaim.where(
+        gene_claim_id: gene_claim.id,
+        drug_claim_id: drug_claim.id,
+        source_id: source.id,
+        known_action_type: 'n/a',
+      ).first_or_create.tap do |interaction_claim|
```

Comment:
The `tap` isn't strictly needed here though it will work just fine as written! `first_or_create` itself takes an optional block, and that block is only called when a new record is created so unless we expect to have duplicate interaction claims with differing attributes, we can skip the `tap`